### ISSUE TITLE
AttributeError: 'list' object has no attribute 'size'

### ISSUE THREAD

**Issue by InstantWindy** [dbolya/yolact#243]
![image](https://user-images.githubusercontent.com/23608180/70377129-c9606200-194b-11ea-8d51-c3a9399afda5.png)

**Comment by dbolya** [comment]
I'm going to need some more context here. It looks like what was passed into Yolact was a list not a pytorch Tensor. Are you writing some custom code?

**Comment by InstantWindy** [comment]
No. I train it on a single GPU. I just uncomment 'net = CustomDataParallel(NetLoss(net, criterion))'.

**Comment by dbolya** [comment]
Why exactly did you comment that out? It does the data preparation so it needs to be run. It's needed even if you only have 1 GPU.

**Comment by InstantWindy** [comment]
Yeah. Thanks. This function 'prepare_data' convert data of type list to data of tensor?

**Comment by dbolya** [comment]
Yeah pretty much. It also splits the data into batches for each GPU, and in the 1 GPU case just spits out arrays with 1 element each.

**Comment by Ahme1994** [comment]
Hello ,can someone helps me ,when I run my code I got this error 
AttributeError: 'list' object has no attribute 'size'
#rpm
for epoch in range(300):
    for x,y in enumerate(train_loader):
        y = torch.zeros(y.size(0),4).scatter_(1,y.view(-1,1),1.)
        x,y = x.to(device),y.to(device)
        output,_ = rpm_nn(x,y)              # cnn output
        loss = loss_func(output)   # cross entropy loss# clear gradients for this training step
        optimizer_rpm.zero_grad()
        loss.backward()
        optimizer_rpm.step()
    test_out,_=rpm_nn(test_x,test_y)
    pred_z=torch.max(test_out,1)[1].cpu().data.numpy()
    accuracy = float((pred_z == test_z.cpu().numpy()).astype(int).sum()) / float(test_y.size(0)) 
    print('Epoch: ', epoch, '| train loss: %.4f' % loss.cpu().data.numpy(), '| test accuracy: %.2f' % accuracy)
